nodedb.go

```diff
@@ -113,6 +120,40 @@ func (ndb *nodeDB) GetNode(hash []byte) *Node {
 	return node
 }
 
+func (ndb *nodeDB) GetFastNode(key []byte) (*FastNode, error) {
+	ndb.mtx.Lock()
+	defer ndb.mtx.Unlock()
```

Comment:
We can do this as RLock

Reply:
On [L134](https://github.com/cosmos/iavl/pull/447/files#diff-6df49839605c94e5caeb7d4549afe697ad97a00144ef26ec6746d9b5e8a3019cR134) there is the chance that the cache-queue is updated which would still need to be protected for writes

Reply:
oh I see. We should probably add a note to make a second Write lock just for that queue later then. (Difference between an RLock and WLock for performance here will be notable, since we want/need parallel queries)

Can write a TODO for this, and then come back to it later?

Reply:
Sounds good to me, added a TODO to circle back around to later.

Reply:
we can move the mutex operation to line 130